pettingzoo/classic/checkers/checkers.py

```diff
@@ -0,0 +1,608 @@
+        dest_pos = 0
+        if direction == 0:
+            # Move up-left
+            dest_pos = pos - 9
+
+            if check_jump(dest_pos):
+                dest_pos = dest_pos - 9
+        elif direction == 1:
+            # Move up-right
+            dest_pos = pos - 7
+
+            if check_jump(dest_pos):
+                dest_pos = dest_pos - 7
+        elif direction == 2:
+            # Move down-left
+            dest_pos = pos + 7
+
+            if check_jump(dest_pos):
+                dest_pos = dest_pos + 7
+        elif direction == 3:
+            # Move down-right
+            dest_pos = pos + 9
+
+            if check_jump(dest_pos):
+                dest_pos = dest_pos + 9
+        else:
+            print("Invalid direction {}".format(direction))
+
+        return (self._act_abs_to_rel(pos), self._act_abs_to_rel(dest_pos))
+
+    def legal_moves(self):
+        moves = self.ch.legal_moves()
+        legal_moves = []
+        for move in moves:
+            srcpos = self._act_rel_to_abs(move[0])
+            destpos = self._act_rel_to_abs(move[1])
+
+            direction = -1
+            if destpos == srcpos - 9 or destpos == srcpos - 18:
+                direction = 0
+            elif destpos == srcpos - 7 or destpos == srcpos - 14:
+                direction = 1
+            elif destpos == srcpos + 7 or destpos == srcpos + 14:
+                direction = 2
+            elif destpos == srcpos + 9 or destpos == srcpos + 18:
+                direction = 3
+
+            legal_moves.append(self._act_abs_to_rel(srcpos) + (64 * direction))
+
+        return legal_moves
+
+    def step(self, action, observe=True):
+
+        if action not in self.legal_moves():
+            warnings.warn(
+                "Bad checkers move made, game terminating with current player losing. \n env.infos[player]['legal_moves'] contains a list of all legal moves that can be chosen."
+            )
+            winner = "white" if self.last_turn == "black" else "black"
+        else:
+            self.num_moves += 1
+            action = self._parse_action(action)
+            self.board, turn, last_moved_piece, moves, winner = self.ch.move(
+                action[0], action[1]
+            )
+
+            self.agent_selection = self._agent_selector.next()
+
+            if turn == self.last_turn:
+                self.agent_selection = self._agent_selector.next()
+                self.agent_order.reverse()
+            self.last_turn = turn
+
+            self.observation = self._read_observation()
+
+            # self.ch.print_board()
+            # print(self.agent_selection)
+            # print(self.observe(self.agent_selection))
+            # print(self.observation)
+            # self.infos[self.agent_selection]['legal_moves']
+
+        self.infos[self.agent_selection]["legal_moves"] = self.legal_moves()
+
+        if winner is None and self.num_moves > self.num_moves_max:
+            print("Draw")

```

Comment:
please remove the print statement. There is actually a test to catch print statement in the CI tests, but those aren't running for some reason.

Reply:
Sure